File: lib/models/farmer.dart
class Farmer {
  final String id;
  final String dairyCode;
  final String name;
  final String nickname;
  final String mobile;
  final double lastLiters;

  const Farmer({
    required this.id,
    required this.dairyCode,
    required this.name,
    required this.nickname,
    required this.mobile,
    this.lastLiters = 0.0,
  });

  Farmer copyWith({
    String? id,
    String? dairyCode,
    String? name,
    String? nickname,
    String? mobile,
    double? lastLiters,
  }) {
    return Farmer(
      id: id ?? this.id,
      dairyCode: dairyCode ?? this.dairyCode,
      name: name ?? this.name,
      nickname: nickname ?? this.nickname,
      mobile: mobile ?? this.mobile,
      lastLiters: lastLiters ?? this.lastLiters,
    );
  }
}

File: backend/models/Farmer.js
const mongoose = require('../mockMongoose');

const farmerSchema = new mongoose.Schema({
  id: { type: String, required: true }, // Local ID like F001
  dairyCode: { type: String, required: true },
  name: { type: String, required: true },
  mobile: { type: String },
  createdAt: { type: Date, default: Date.now },
  updatedAt: { type: Date, default: Date.now }
});

// Composite unique index
farmerSchema.index({ dairyCode: 1, id: 1 }, { unique: true });

module.exports = mongoose.model('Farmer', farmerSchema);
